### PR TITLE
Update types/node16 to 20 and v3 in readmes

### LINKED ISSUE #1323: Updated to actions/cache@v4, but that does not stop the Node16 warning on v3
https://github.com/xenserver/status-report/actions/runs/7802115676

This workflow (you can check the workflow file in the job itself) is upgraded `actions/cache@v4`, but that does not stop the warning about @v3...:

```py
Python3: Pre-Commit Suite
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/cache@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```

### PR DESCRIPTION
**Pull request by bernhardkaindl**
Untested (not a js dev), but should not need any remarks else than:

Commit 1 may be one step forward to help to fix #1323, best wishes to remove all node16 deps.

But that does not explain why the GitHub annotation still talks  that it is cache@v3 and not v4...

yargs also uses a 16.x.x number in package-lock.json, setup-python uses a 17.x.x version, maybe it and others need updating too

### CODE REVIEW

**Review by ulgens — APPROVED** [review]

**Review by martibenn — APPROVED** [review]

**Comment by bernhardkaindl** [comment]
@bethanyj28, maybe you might be able to take a look at this and at the confirmed #1323?

File: package-lock.json [hunk 1]
```diff
@@ -11816,8 +11816,8 @@
       }
     },
     "@types/node": {
-      "version": "16.18.3",
-      "resolved": "https://registry.npmjs.org/@types/node/-/node-16.18.3.tgz",
+      "version": "20.9.1",
+      "resolved": "https://registry.npmjs.org/@types/node/-/node-20.9.1.tgz",
```
Comment (bernhardkaindl) [inline package-lock.json:11820]: I hope this may fix #1323

**Review by ShenMian — APPROVED** [review]

File: examples.md [hunk 2]
```diff
@@ -46,7 +46,7 @@
 Using [NuGet lock files](https://docs.microsoft.com/nuget/consume-packages/package-references-in-project-files#locking-dependencies):
 
 ```yaml
-- uses: actions/cache@v3
+- uses: actions/cache@v4
```
Comment (hamirmahal) [inline examples.md:49]: https://github.com/actions/cache/pull/1378/files#diff-cfb2db6f153a4484d9bbd408f3227a264dce68fadd747cecdc30bda012b4d8cdR49 also makes this change.

**Comment by IanButterworth** [comment]
This repo seems to be lacking maintenance attention. It may be helpful to make some noise on here https://github.com/orgs/community/discussions/121811

File: examples.md [hunk 3]
```diff
@@ -55,10 +55,10 @@ Using [NuGet lock files](https://docs.microsoft.com/nuget/consume-packages/packa
 ```
 
 Depending on the environment, huge packages might be pre-installed in the global cache folder.
-With `actions/cache@v3` you can now exclude unwanted packages with [exclude pattern](https://github.com/actions/toolkit/tree/main/packages/glob#exclude-patterns)
+With `actions/cache@v4` you can now exclude unwanted packages with [exclude pattern](https://github.com/actions/toolkit/tree/main/packages/glob#exclude-patterns)
```
Comment (joshmgross) [inline examples.md:58]: This feature was added in `v3` - 
```suggestion
With `v3` and higher, you can now exclude unwanted packages with [exclude pattern](https://github.com/actions/toolkit/tree/main/packages/glob#exclude-patterns)
```

**Comment by IanButterworth** [comment]
bump. At least run CI?

Leaving this action like this seems very counter to the whole github actions push to update to node 20.

**Comment by joshmgross** [comment]
Since https://github.com/actions/cache/issues/1323#issuecomment-2260983836 has already been resolved and was an unrelated issue, I don't think we need these changes.

Feel free to reopen if you feel otherwise.

**Pull request closed without merging**